core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -2334,3 +2335,26 @@ strip eq arr@Array{..} = do
             then getLast prev low
             else return cur
         else return cur
+
+-- | Bubble sorts the elements.
+--
+-- /Pre-release/
+{-# INLINE bubbleAsc #-}
+bubbleAsc :: (MonadIO m, Storable a) =>
+    Int -> (a -> a -> Ordering) -> Array a -> m ()
+bubbleAsc idx cmp0 arr =
```

Comment:
We will usually be appending elements to the array. In this API we can assume that the out-of-order element is the last element of the array. So we need not pass idx.